### DISCUSSION TITLE
Differential Privacy on tree based models[Q&A]

### DISCUSSION THREAD

**Discussion by ammar-sandboxaq** [NVIDIA/NVFlare#2320] [Q&A] [answered]
### Python version (`python3 -V`)

Python 3.9.2

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

nvflare-nightly           2.4.0rc5+36.g21cc1000 

### Operating system

Debian GNU/Linux 11

### Have you successfully run any of the following examples?

- [ ] hello-numpy-sag with simulator
- [X] hello-pt with simulator
- [ ] hello-numpy-sag with POC
- [ ] hello-pt with POC

### Please describe your question

I have two high level questions.
1-) Tree based models does not work with stable pip versions but only works with nightly version using setup.py. Do you plan to update the stable version for tree-based models. The problem I am having is with `FedXGBTreeExecutor`.

2-) I have been experimenting nvflare with tree based models on a custom dataset without any issue. However, when I try to implement DP as described in [brast18](https://github.com/NVIDIA/NVFlare/tree/03af6b6ba0c2f9bf3145f492e773946a2aa88699/examples/advanced/brats18#differential-privacy-dp) by updating the config_fed_client.json file in the `jobs/`, the performance does not change at all. 

I tried all three methods `Histogram`, `Bagging` and `Cyclic` with different noise parameters but it seems `SVTPrivacy` is not executed during training. Is there anything I am missing? Can you please enlighten me regarding this. 

Note: To verify if SVTPrivacy works, I tested hello-pt example with it and it works without any problem. 

Thank you

**Comment by YuanTingHsieh** [comment]
For the first question, if you are using NVFlare version 2.3, please use examples from branch 2.3: https://github.com/NVIDIA/NVFlare/tree/2.3

For the second question, that is interesting, @ZiyueXu77 can you share some thoughts on this? thanks

**Reply by ZiyueXu77** [reply, marked as answer]
@ammar-sandboxaq, SVTPrivacy currently works only for deep learning models and does not directly apply to xgboost models due to two reasons:
1. for histogram and cyclic, at this time NVFlare serves as a simple bridge to connect xgboost's distributed frame, such that nvflare itself do not have the capability of adding noise to the message.
2. for bagging, what gets communicated is the model itself with a special tree structure, hence it may not make much sense to add DP in such case. Even if assuming DP is necessary, special design and implementation will be needed to adapt SVTPrivacy.
